include/nil/crypto3/zk/snark/systems/plonk/placeholder/prover.hpp

```diff
@@ -236,7 +236,21 @@ namespace nil {
 
                         PROFILE_PLACEHOLDER_SCOPE("split_polynomial_dfs_conversion_time");
 
-                        std::vector<polynomial_dfs_type> T_splitted_dfs(T_splitted.size());
+                        std::size_t split_polynomial_size = std::max(
+                            (preprocessed_public_data.identity_polynomials.size() + 1) * (preprocessed_public_data.common_data.rows_amount -1 ),
+                            (constraint_system.lookup_poly_degree_bound() + 1) * (preprocessed_public_data.common_data.rows_amount -1 )//,
+                        );
+                        split_polynomial_size = std::max(
+                            split_polynomial_size,
+                            (preprocessed_public_data.common_data.max_gates_degree + 1) * (preprocessed_public_data.common_data.rows_amount -1)
+                        );
+                        split_polynomial_size = (split_polynomial_size % preprocessed_public_data.common_data.rows_amount != 0)?
+                            (split_polynomial_size / preprocessed_public_data.common_data.rows_amount + 1):
+                            (split_polynomial_size / preprocessed_public_data.common_data.rows_amount);
+

```

Comment:
Can you please add a comment somewhere here describing why we want the sizes to be equal, so later we don't forget.